lib/LedgerSMB/Template/DB.pm

```diff
@@ -131,12 +131,15 @@ does not call this without carefully whitelisting values.
 
 sub get_from_file {
     my ($package, $path, $language_code) = @_;
-    my $fname = $path;
+    my $fname;
     if ($path =~ m|/.*:| ){
        die 'Cannot run on NTFS alternate data stream!';
     }
-    $path =~ m|(.*)/([^/]+)$|;
-    $fname = $2;
+    if ( $path =~ m|(.*)/([^/]+)$| ) {
+        $fname = $2;
+    }  else {
+        $fname = $path;
+    }
```

Comment:
I agree with your analysis.  And I think this change is worse that you do.  I'll rework it.